### DISCUSSION TITLE
minor change of peer & dev dependency causes major change.

### DISCUSSION THREAD

**Discussion by emewjin** [changesets/changesets#988] [Q&A]
Hi, I got problem that minor change of peer & dev dependency causes major change.
I already see [similar issue](https://github.com/changesets/changesets/issues/524) and try to add experimental config like below, but it still in trouble.
```
{
  "$schema": "https://unpkg.com/@changesets/config@2.1.1/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "fixed": [],
  "linked": [],
  "access": "restricted",
  "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "ignore": [],
  "___experimentalUnsafeOptions_WILL_CHANGE_IN_PATCH": {
    "onlyUpdatePeerDependentsWhenOutOfRange": true
  }
}
```

Unfortunately the repo is private because of owner's decision, so I please ask for your understanding about just describe problem by text.

There is package A and B. 
A depends on B (peer&dev).
If I make minor change on B, A got major changes.
Here is screenshot that changeset bot made.
![Screenshot_20221025_011730](https://user-images.githubusercontent.com/76927618/197575975-a5682af0-3484-41cf-8a1e-4ba35ce090cc.png)
This is minor change on B.

![Screenshot_20221025_011937](https://user-images.githubusercontent.com/76927618/197577160-6febf70b-4c0a-450e-8c95-a638737ac716.png)
This is major change on A.

Why minor change of peer dependency causes major change even if I set `onlyUpdatePeerDependentsWhenOutOfRange` true?
Did I something wrong in config?

**Comment by unional** [comment]
https://github.com/justland/just-web/pull/130

I notice the same issue.
I recall the setting I have used to solve that problem.

Regression?

```
{
  "$schema": "https://unpkg.com/@changesets/config@2.0.0/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "title": "chore: version packages",
  "fixed": [
    [
      "@just-web/app",
      "@just-web/browser-keyboard",
      "@just-web/browser-preferences",
      "@just-web/browser",
      "@just-web/commands",
      "@just-web/events",
      "@just-web/keyboard",
      "@just-web/log",
      "@just-web/os",
      "@just-web/preferences",
      "@just-web/routes",
      "@just-web/states",
      "@just-web/types"
    ]
  ],
  "linked": [],
  "access": "public",
  "baseBranch": "main",
  "___experimentalUnsafeOptions_WILL_CHANGE_IN_PATCH": {
    "onlyUpdatePeerDependentsWhenOutOfRange": true,
    "updateInternalDependents": "always"
  },
  "updateInternalDependencies": "patch",
  "ignore": []
}
```

**Reply by unional** [reply]
I've confirmed that then the dependency have a `minor` change, the dependent got an major. with this setting.

e.g. `minor` change for `@just-web/react` causes `major` change for `@just-web/react-commands`.

Seems like the workaround only works for `patch` changes.

**Comment by Andarist** [comment]
@emewjin Please always try to share a repro case in a runnable form - providing a git repository to clone should be OK. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given. I can't possibly know what might be wrong in your case without the ability to inspect a sample repo.

@unional your situation is related to using `workspace` ranges and to the fact that they are not correctly handled here:
https://github.com/changesets/changesets/blob/1d3f9f29f4cfe3c32d18a98a1dc16b1bfce9e7e6/packages/assemble-release-plan/src/determine-dependents.ts#L193-L197

I actually have a pending fix for this in a form of a draft PR:
https://github.com/changesets/changesets/pull/975/files#diff-228f979d49ecba43873ed35ecce890137f48f0021cb3bb21fa4d1312f3c15351R206-R207

I plan to get back to this draft soon - the only thing that I need there is writing tests for fixed cases.

**Reply by jmcdo29** [reply]
In the meantime with this, should `workspace: *` be used to skip the major updates of peerDeps?

**Comment by emewjin** [comment]
@Andarist I noticed that this situation was exact work by check [this documentation](https://github.com/changesets/changesets/blob/main/docs/decisions.md#the-versioning-of-peer-dependencies). This was not a bug 🙏🏻
